src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void createNewArray() {
+        int newSize = array.length + array.length / 2;
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
+    }
```

Comment:
The method `createNewArray` should be renamed to reflect that it's conditionally resizing the array, and the resizing logic should be in a separate method according to the Single Responsibility principle. Consider splitting the logic into a method that checks if resizing is needed and another that actually performs the resizing.